app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt

```diff
@@ -171,6 +166,24 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
     })
   }
 
+  private fun showAllowMobileNetworkDialog() {
+    dialogShower.show(WifiOnly, {
+      sharedPreferenceUtil.putPrefWifiOnly(false)
+      zimManageViewModel.shouldShowWifiOnlyDialog.value = false
+    }, {
+      dialogShower.show(
+        KiwixDialog.AllowMobileNetwork,
+        ::showAllowMobileNetworkDialog, {
+          onRefreshStateChange(false)
```

Comment:
Here we are calling recursively, it's a bad way to fix this. It'll work but in a bad way!

Reply:
I have remove this condition , now permission dialog will prompt on download navigation every time unless user allow it.